scripts/dts/edtsdatabase.py

```diff
@@ -76,6 +76,20 @@ def insert_device_type(self, compatible, device_type):
             self._edts['device-types'][device_type].append(compatible)
             self._edts['device-types'][device_type].sort()
 
+    def insert_device_controller(self, controller_type, controller, device_id, line):
```

Comment:
This needs some work, as we kinda assume which property of the controller 'irq' or 'pin' we should "sort" the controller view by.  However, it should probably be more generic than that.  Clocks is a good example since we are sorting by some field that isn't consistent across everyone.

Reply:
So my understanding is that we should set `line` as a list, so we're able to inject `device_id` under `bus/bit` path or whatever `xxx\...\zzz` path is needed.
Would that be generic enough?

Reply:
Not sure I follow, what will setting `line` as a list do?

Reply:
`line` is used as nesting parameter.
Today, it is a variable. hence we're constrained to nesting depth of 1 (irq, pin, bus) between controller and device:
Today we have:
for line = pin number:
```
"gpio-controller": gpioa/pin1/devicex
                                    /pin2/devicey
```
for line = irq number
```
"interrupt-controller": interupt-controllera/irq31/devicex
                                                               /irq32/devicey
```

Setting `line` as a list would allow to have a variable nesting depth and allow to have for instance:
line = bus,bit
```
"clock-controller": gpioa/busa/bit1/devicex
                                     /busa/bit2/devicey
```

So each controller could use its own nesting, making the function more generic.

Reply:
That all makes sense, just curious how we handle 2 different clock-controllers.  Lets say one clock-controller defines one cell as 'bits', and second clock-controller defines one say as 'name'.  How will that end up working?

Reply:
Different compatible will handle that.

Reply:
Since we give following arguments (controller_type, controller, device_id, line) we're able to have different stuff for clock-controller Y and clock-controller Z